dft.cpp

```diff
@@ -0,0 +1,59 @@
+#include "fftw.h"
+
+#include <Halide.h>
+#include <fftw3.h>
+
+#include <cassert>
+#include <iostream>
+
+#ifdef _WIN32
+#define DLLEXPORT __declspec(dllexport)
+#else
+#define DLLEXPORT
+#endif
+
+using namespace std;
+
+#ifdef NO_MANGLE
+extern "C" DLLEXPORT
+#endif
```

Comment:
With the recently added `c_plus_plus_name_mangling` feature we shouldn't need this (or the `DLLEXPORT` defined above). Why do we need the `NO_MANGLE` check?

Reply:
This one is easy:

    Until CASPER compiler supports controlling the demangling
    feature of Halide when it invokes the generators.
    
    No time to try to implement this in Casper compiler when we can just have the macro.